include/aspect/volume_of_fluid/utilities.h

```diff
@@ -0,0 +1,152 @@
+/*
+ Copyright (C) 2016 by the authors of the ASPECT code.
+
+ This file is part of ASPECT.
+
+ ASPECT is free software; you can redistribute it and/or modify
+ it under the terms of the GNU General Public License as published by
+ the Free Software Foundation; either version 2, or (at your option)
+ any later version.
+
+ ASPECT is distributed in the hope that it will be useful,
+ but WITHOUT ANY WARRANTY; without even the implied warranty of
+ MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ GNU General Public License for more details.
+
+ You should have received a copy of the GNU General Public License
+ along with ASPECT; see the file doc/COPYING.  If not see
+ <http://www.gnu.org/licenses/>.
+ */
+
+#ifndef _aspect__volume_of_fluid_utilities_h
+#define _aspect__volume_of_fluid_utilities_h
+
+#include <deal.II/base/point.h>
+
+#include <aspect/global.h>
+
+namespace aspect
+{
+  namespace VolumeOfFluid
+  {
+    using namespace dealii;
+
+    // Utilities
+
+    /**
+     * Function to calculate volume fraction contained by indicator function
+     * H(d-normal*(x'-x_{cen}')) on the [0, 1]^dim unit cell where x_{cen} is
+     * the unit cell center.
+     *
+     * Currently only works assuming constant Jacobian determinant.
+     */
+    template<int dim>
+    double compute_fluid_fraction (const Tensor<1, dim, double> normal,
+                                   const double d);
```

Comment:
d -> distance?

Reply:
In effect yes, but in practice d is more accurate due being defined in terms of the $d$ value required for the line $normal dot x = d$ to match the interface.